app/uk/gov/hmrc/gform/testonly/SnapshotADTs.scala

```diff
@@ -16,78 +16,166 @@
+  def apply(snapshotItem: SnapshotItem): SnapshotWithData =
+    SnapshotWithData(Snapshot(snapshotItem), snapshotItem.originalForm.formData)
   implicit val writes: OWrites[SnapshotWithData] = derived.owrites()
 }
 
 case class SaveRequest(
-  formId: String,
-  description: String
+  formId: FormId,
+  description: Description,
+  gformFrontendVersion: GformFrontendVersion
 )
 
 object SaveRequest {
   implicit val format: OFormat[SaveRequest] = derived.oformat()
 }
 
 case class SaveReply(
-  snapshotId: String
+  formId: FormId
 )
 
 object SaveReply {
   implicit val format: OFormat[SaveReply] = derived.oformat()
 }
 
 case class UpdateSnapshotRequest(
-  snapshotId: String,
+  snapshotId: SnapshotId,
   formData: JsObject,
-  description: String
+  description: Description
 )
 
 object UpdateSnapshotRequest {
   implicit val format: OFormat[UpdateSnapshotRequest] = derived.oformat()
 }
 
 case class UpdateFormDataRequest(
-  formId: String,
+  formId: FormId,
   formData: JsObject
 )
 
 object UpdateFormDataRequest {
   implicit val format: OFormat[UpdateFormDataRequest] = derived.oformat()
 }
+case class SnapshotId(value: String) extends AnyVal
+
+object SnapshotId {
+  implicit val format: Format[SnapshotId] = Json.valueFormat
+}
+
+case class Description(value: String) extends AnyVal
+object Description {
+  implicit val format: Format[Description] = Json.valueFormat
+}
+
+case class GformVersion(value: String) extends AnyVal
+
+object GformVersion {
+  implicit val format: Format[GformVersion] = Json.valueFormat
+}
+
+case class GformFrontendVersion(value: String) extends AnyVal
+
+object GformFrontendVersion {
+  implicit val format: Format[GformFrontendVersion] = Json.valueFormat
+}
+
+case class SnapshotItem(
+  snapshotId: SnapshotId,
+  originalForm: Form,
+  originalTemplate: FormTemplateRaw,
+  description: Description,
+  gformVersion: GformVersion,
+  gformFrontendVersion: GformFrontendVersion,
+  snapshotTemplateId: FormTemplateId,
+  createdAt: Instant
+) {
+  def toSnapshotForm(currentForm: Form): Form = {
+    val currentUserId = currentForm.userId
+    val currentId = currentForm._id
+    originalForm.copy(
+      _id = currentId,
+      userId = currentUserId,
+      formTemplateId = snapshotTemplateId
+    )
+  }
+
+  def toSnapshotTemplate(): FormTemplateRaw =
+    originalTemplate.copy(value = originalTemplate.value ++ Json.obj("_id" -> snapshotTemplateId.value))
+
+  def updateWith(newFormData: FormData, newDescription: Description): SnapshotItem =
+    copy(originalForm = originalForm.copy(formData = newFormData), description = newDescription)
+}
+
+object SnapshotItem {
+
+  def apply(
+    originalForm: Form,
+    originalTemplate: FormTemplateRaw,
+    description: Description,
+    gformVersion: GformVersion,
+    gformFrontendVersion: GformFrontendVersion
+  ): SnapshotItem = {
+    // add prefix to the template id for both saved form and template
+    val prefix = UniqueStringGenerator.generateUniqueString()
```

Comment:
Use `snapshotId` bit instead.

That will have an added benefit that we will be able to identified what snapshot was current session created from (it can come handy)